tests/e2escenarios/e2e_test.go

```diff
@@ -259,6 +259,230 @@ var _ = Describe("E2E Test", func() {
+			getSVCArgs := []string{"get", "svc", "-n", commonVar.Project}
+
+			command := []string{"odo", "init"}
+			_, err := helper.RunInteractive(command, nil, func(ctx helper.InteractiveContext) {
+
+				// helper.ExpectString(ctx, "Based on the files in the current directory odo detected")
+				helper.ExpectString(ctx, "Language: javascript")
+				helper.ExpectString(ctx, "Project type: nodejs")
+				helper.ExpectString(ctx, "Is this correct")
+
+				helper.SendLine(ctx, "\n")
+
+				helper.ExpectString(ctx, "Select container for which you want to change configuration?")
+
+				helper.SendLine(ctx, "\n")
+
+				helper.ExpectString(ctx, "Enter component name")
+
+				helper.SendLine(ctx, componentName)
+
+				helper.ExpectString(ctx, "Your new component '"+componentName+"' is ready in the current directory")
+
+			})
+			Expect(err).To(BeNil())
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("devfile.yaml"))
+
+			//add binding information (binding as file)
+			helper.Cmd("odo", "add", "binding", "--name", bindingName, "--service", "cluster-sample").ShouldPass()
+
+			// "execute odo dev and add changes to application"
+			var devSession helper.DevSession
+			var ports map[string]string
+
+			devSession, _, _, ports, err = helper.StartDevMode(nil)
+			helper.ReplaceString(filepath.Join(commonVar.Context, "server.js"), "from Node.js", "from updated Node.js")
+			Expect(err).ToNot(HaveOccurred())
+
+			_, _, _, err = devSession.WaitSync()
+			Expect(err).ToNot(HaveOccurred())
+
+			// "should update the changes"
+			checkIfDevEnvIsUp(ports["3000"], "Hello from updated Node.js Starter Application!")
+
+			// check odo describe to check for env
+			stdout := helper.Cmd("odo", "describe", "binding", "-o", "json").ShouldPass().Out()
+			Expect(helper.IsJSON(stdout)).To(BeTrue())
+			helper.JsonPathContentIs(stdout, "0.name", bindingName)
+			helper.JsonPathContentIs(stdout, "0.spec.application.kind", "Deployment")
+			helper.JsonPathContentIs(stdout, "0.spec.application.name", componentName+"-app")
+			helper.JsonPathContentIs(stdout, "0.spec.application.apiVersion", "apps/v1")
+			helper.JsonPathContentIs(stdout, "0.spec.services.0.apiVersion", "postgresql.k8s.enterprisedb.io/v1")
+			helper.JsonPathContentIs(stdout, "0.spec.services.0.kind", "Cluster")
+			helper.JsonPathContentIs(stdout, "0.spec.services.0.name", "cluster-sample")
+			helper.JsonPathContentIs(stdout, "0.spec.detectBindingResources", "true")
+			helper.JsonPathContentIs(stdout, "0.spec.bindAsFiles", "true")
+			helper.JsonPathContentContain(stdout, "0.status.bindingFiles", fmt.Sprintf("%q", "${SERVICE_BINDING_ROOT}/"+bindingName+"/password"))
+
+			// check if secrets are mounted in
+			podName := commonVar.CliRunner.GetRunningPodNameByComponent(componentName, commonVar.Project)
+			stdout = commonVar.CliRunner.Exec(podName, commonVar.Project, "bash", "-c", "ls", "'${SERVICE_BINDING_ROOT}/"+bindingName+"/password'")
```

Comment:
To clarify my comment above about a more realistic project, I think this would help avoid testing binding this way. This is OK, but IMO, for an end-to-end test, we should check that the application (which is expected to consume the binding information) responds correctly. What do you think?